sbt-bridge/src/xsbt/CompilerClassLoader.scala

```diff
@@ -42,6 +44,14 @@ class CompilerClassLoader(urls: Array[URL], sbtLoader: ClassLoader)
 }
 
 object CompilerClassLoader {
+  /** Cache the result of `fixBridgeLoader`.
+   *
+   *  Reusing ClassLoaders is important for warm performance since otherwise the
+   *  JIT code cache for the compiler will be discarded between every call to
+   *  the sbt `compile` task.
+   */
+  private[this] val fixedLoaderCache = new mutable.WeakHashMap[ClassLoader, ClassLoader]
```

Comment:
in weak hashMap only values are weak. You are keeping SBT classloaders(keys) from being GC-d.
I'd prefer to have one more level of indirection to also remove this issue.